mypy/checkexpr.py

```diff
@@ -411,7 +411,9 @@ def analyze_ref_expr(self, e: RefExpr, lvalue: bool = False) -> Type:
             result = self.alias_type_in_runtime_context(
                 node, ctx=e, alias_definition=e.is_alias_rvalue or lvalue
             )
-        elif isinstance(node, (TypeVarExpr, ParamSpecExpr, TypeVarTupleExpr)):
+        elif isinstance(node, TypeVarExpr):
+            return self.named_type("typing.TypeVar")
```

Comment:
Will this cause crashes on Python versions where `TypeVar` is not a proper class? And/or if it is imported from `typing_extensions` (e.g. by people who want the type variables with defaults)?

Reply:
TypeVar has been a class since at least 3.6, for what it's worth: https://github.com/python/cpython/blob/8d999cbf4adea053be6dbb612b9844635c4dfb8e/Lib/typing.py#L453.

Reply:
`TypeVar` is defined unconditionally as a class in typeshed:
```py
@final
class TypeVar:
    @property
    def __name__(self) -> str: ...
    @property
    def __bound__(self) -> Any | None: ...
    ...
```
Also on the oldest supported Python version (3.8) `TypeVar` is a class:
```
Python 3.8.10 (v3.8.10:3d8993a744, May  3 2021, 09:09:08)
[Clang 12.0.5 (clang-1205.0.22.9)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from typing import TypeVar
>>> TypeVar
<class 'typing.TypeVar'>
```
But this won't be right if `TypeVar` is imported from `typing_extensions`. It probably won't usually make a difference, but it's possible to imagine runtime introspection use cases where mypy could get confused. But this wasn't properly supported before (when the type was `object`), so it doesn't seem like a major issue. I may fix this in a follow-up PR if it's easy enough.